simplejson/simplejson.go

```
@@ -0,0 +1,268 @@
+	labels := ""
+	metricName := ""
+	for _, column := range frame.Columns {
+		if column.Name != "values" {
+			values, _ := req.extractColumnValuesTSDB(column)
+			val := fmt.Sprintf("%s", values.Index(0).Interface())
+			if column.Name == "metric_name" {
+				metricName = val
+			} else {
+				labels += fmt.Sprintf("[%s=%s]", column.Name, val)
+			}
+		}
+	}
+	return metricName + labels
+}
+
+func (req *SimpleJsonQueryRequest) CreateResponse(iter frames.FrameIterator) (interface{}, error) {
+	formatters := map[string]func(iter frames.FrameIterator) (interface{}, error){"kv": req.formatKV, "tsdb": req.formatTSDB}
+	if formatter, ok := formatters[req.Backend]; ok {
+		return formatter(iter)
+	}
+	return nil, fmt.Errorf("Unknown format: %s", req.Backend)
+}
+
+func (req *SimpleJsonSearchRequest) CreateResponse(iter frames.FrameIterator) (interface{}, error) {
+	// Placeholder for actual implementation
+	return []string{}, nil
+}
+
+func (req *SimpleJsonQueryRequest) ParseRequest(requestBody []byte) error {
+	if err := json.Unmarshal(requestBody, req); err != nil {
+		return err
+	}
+	for _, target := range req.Targets {
+		req.Type = target["type"].(string)
+		fieldInput := target["target"].(string)
+		if err := req.parseQueryLine(fieldInput); err != nil {
+			return err
+		}
+	}
+	req.parseRange()
+	return nil
+}
+
+func (req *SimpleJsonSearchRequest) ParseRequest(requestBody []byte) error {
+	if err := json.Unmarshal(requestBody, req); err != nil {
+		return err
+	}
+	for _, target := range strings.Split(req.Target, QUERY_SEPARATOR) {
+		fmt.Println(target)
+		if err := req.parseQueryLine(strings.TrimSpace(target)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (req *SimpleJsonQueryRequest) parseQueryLine(fieldInput string) error {
+	translate := map[string]string{"table_name": "TableName"}
+	re := regexp.MustCompile(`^\s*(filter|fields|table_name|backend|container|username|password)\s*=\s*(.*)\s*$`)
```

This should probably be in `init` so we'll crash before we start the server if the regexp is not valid. (fail fast)

---

This (query) is part of the request payload

---

I mean the `re.MustCompile` which might panic on invalid regexp

---

worst error I got is "failed to create adapter with a random string, but I have replaced it with "Compile" for better error handling